builder/openstack/step_create_volume.go

```diff
@@ -79,15 +78,14 @@ func (s *StepCreateVolume) Run(ctx context.Context, state multistep.StateBag) mu
 	// Wait for volume to become available.
 	ui.Say(fmt.Sprintf("Waiting for volume %s (volume id: %s) to become available...", config.VolumeName, volume.ID))
 	if err := WaitForVolume(blockStorageClient, volume.ID); err != nil {
+		// Put the error volume here for cleanup.
+		s.volumeID = volume.ID
```

Comment:
I would move this after the call to `Create`, assuming there wasn't an error immediately, the volume is (I assume) created and in need of a cleanup afterwards.
Since it was already set below (l.92), and we also add this here, we can mutualise this assignation so we don't need to do it twice.